plugins/modules/foreman_host.py

```diff
@@ -86,76 +116,48 @@
 
 RETURN = ''' # '''
 
-try:
-    from nailgun.entities import (
-        Host,
-        HostGroup,
-    )
-
-    from ansible.module_utils.ansible_nailgun_cement import (
-        find_host,
-        find_hostgroup,
-        find_location,
-        find_organization,
-        ForemanEntityAnsibleModule,
-        naildown_entity_state,
-        sanitize_entity_dict,
-    )
-except ImportError:
-    pass
-
-
-# This is the only true source for names (and conversions thereof)
-name_map = {
-    'name': 'name',
-    'enabled': 'enabled',
-    'managed': 'managed',
-    'hostgroup': 'hostgroup',
-    'location': 'location',
-    'organization': 'organization',
-}
+from ansible.module_utils.foreman_helper import ForemanEntityAnsibleModule
 
 
 def main():
     module = ForemanEntityAnsibleModule(
-        argument_spec=dict(
+        entity_spec=dict(
             name=dict(required=True),
-            hostgroup=dict(),
-            location=dict(),
-            organization=dict(),
-            enabled=dict(default='true', type='bool'),
+            hostgroup=dict(type='entity', flat_name='hostgroup_id'),
+            location=dict(type='entity', flat_name='location_id'),
+            organization=dict(type='entity', flat_name='organization_id'),
+            enabled=dict(type='bool'),
```

Comment:
just for my understanding. you dropped the default enabled=true, because enabled: true and managed: false make no sense together?

Reply:
my reasoning was: enabled, build and managed are *optional* args in the API and the API will set them to reasonable defaults, so there is no reason to copy this default from Foreman into the module

Reply:
ok, that make sense. thx